bionemo-recipes/recipes/esm2_native_te/dataset.py

```diff
@@ -215,7 +218,158 @@ def create_thd_dataloader(
         collate_fn=data_collator,
         num_workers=num_workers,
         pin_memory=True if not use_stateful_dataloader else False,
-        persistent_workers=True,
     )
-
     return train_dataloader, tokenized_dataset
+
+
+def create_cp_dataloader(
+    distributed_config: DistributedConfig,
+    tokenizer_name: str,
+    load_dataset_kwargs: dict,
+    micro_batch_size: int | None = None,
+    token_micro_batch_size: int | None = None,
+    num_workers: int = 1,
+    max_seq_length: int = 1024,
+    seed: int = 42,
+    buffer_size: int = 10_000,
+    use_stateful_dataloader: bool = False,
+    mlm_probability: float = 0.15,
+    pad_sequences_to_be_divisible_by: int | None = None,
```

Comment:
since you know the cp_world_size, can't we initialize this to the correct value for folks?